nova_vm/src/ecmascript/abstract_operations/operations_on_iterator_objects.rs

```diff
@@ -0,0 +1,103 @@
+//! ## [7.4 Operations on Iterator Objects](https://tc39.es/ecma262/#sec-operations-on-iterator-objects)
+
+use super::{
+    operations_on_objects::{call, get},
+    type_conversion::{to_boolean, to_object},
+};
+use crate::ecmascript::{
+    execution::{agent::ExceptionType, Agent, JsResult},
+    types::{Function, Object, PropertyKey, Value},
+};
+
+/// [7.4.2 GetIteratorFromMethod ( obj, method )](https://tc39.es/ecma262/#sec-getiteratorfrommethod)
+///
+/// The abstract operation GetIteratorFromMethod takes arguments obj (an
+/// ECMAScript language value) and method (a function object) and returns
+/// either a normal completion containing an Iterator Record or a throw
+/// completion.
+pub(crate) fn get_iterator_from_method(
+    agent: &mut Agent,
+    obj: Value,
+    method: Function,
+) -> JsResult<Object> {
+    // 1. Let iterator be ? Call(method, obj).
+    let iterator = call(agent, method.into(), obj, None)?;
+
+    // 2. If iterator is not an Object, throw a TypeError exception.
+    let iterator = if let Ok(iterator) = to_object(agent, iterator) {
+        iterator
+    } else {
+        return Err(agent.throw_exception(ExceptionType::TypeError, "Iterator is not object"));
+    };
```

Comment:
Note that you can just do this:

```suggestion
    let Ok(iterator) = to_object(agent, iterator) else {
        return Err(agent.throw_exception(ExceptionType::TypeError, "Iterator is not object"));
    };
```

Reply:
Yeah, nifty in our project with how we dance around with enums :)

Reply:
I didn't know this either, and indeed VERY! handy! Man, I love this right now. Thank you @sno2 !